### PR TITLE
Remove calls to deprecated CoreAudio API

### PR DESCRIPTION
**Pull request by l-tms**
Remove calls to deprecated CoreAudio API

Resolves: audacity/audacity#1462

I changed all the deprecated CoreAudio API calls to use the current active ones.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: lib-src/portmixer/src/px_mac_coreaudio.c [hunk 1]
```diff
@@ -523,14 +543,20 @@ static PxVolume get_play_through(px_mixer *Px)
    OSStatus err;
    UInt32   outSize;
    UInt32   flag;
+   AudioObjectPropertyAddress inAddress = {
+      kAudioDevicePropertyPlayThru,
+      // should we use ...ScopePlayThrough, ...ScopeGlobal or ...ScopeInput instead?
```
Comment (crsib) [inline lib-src/portmixer/src/px_mac_coreaudio.c:548]: Gosh I live Apple documentation :-)

https://developer.apple.com/documentation/coreaudio/1494464-anonymous/kaudioobjectpropertyscopeplaythrough?language=objc

This is actually an excellent question of which scopes to use here. Have you found a way to check this under the debugger?

Comment (l-tms) [reply lib-src/portmixer/src/px_mac_coreaudio.c:548]: I checked in the debugger the output for all new calls I introduced, comparing them to the old call version, but for this I never managed to get back anything useful. 

Even with the old version I got back `OSStatus 2003332927 kAudioHardwareUnknownPropertyError` every time, with both the MacBook builtin and external soundcard I have at home, so I assumed I was just missing hardware that would support this.

I used `ScopeOutput` in the end as the original code was also using that (`IS_OUTPUT`)

Comment (crsib) [reply lib-src/portmixer/src/px_mac_coreaudio.c:548]: I think that we need to use PlayThrough scope

Some other guys do at least xD

https://github.com/itsthejb/HotPants/blob/master/CoreAudio/HAL/HPBase/HP_Device.cpp

Comment (l-tms) [reply lib-src/portmixer/src/px_mac_coreaudio.c]: fair enough, I updated this line to use `kAudioDevicePropertyScopePlayThrough` :)

**Review by crsib — APPROVED** [review]

**Comment by crsib** [comment]
Thank you a lot!

**Pull request merged**